### ISSUE TITLE
check_http ignores --hostname when --header is used

### ISSUE THREAD

**Issue by koenwtje** [nagios-plugins/nagios-plugins#79]
If the `--header` option is used together with the `--hostname` option, check_http doesn't include the `Host:` header (the `--hostname` option seems to be ignored).

**Comment by sreinhardt** [comment]
Interesting, I'll take a serious look in ~2-3 weeks when other engagements are completed. thanks!

**Comment by tmcnag** [comment]
I was not able to reproduce this behavior:

```
./check_http --version
check_http v2.0.3 (nagios-plugins 2.0.3)
```

Run with just --hostname:

```
./check_http --hostname=www.example.com
HTTP OK: HTTP/1.1 200 OK - 1610 bytes in 0.041 second response time |time=0..040901s;;;0.000000 size=1610B;;;0
```

Associated `tcpdump dst port 80 -A` output (snipped):

```
User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3)
Connection: close
Host: www.example.com
Accept: */*
```

Run with --hostname and a "Cookie" header with --header:

```
./check_http --hostname=www.example.com --header="Cookie: foo=bar"
HTTP OK: HTTP/1.1 200 OK - 1610 bytes in 0.041 second response time |time=0.040508s;;;0.000000 size=1610B;;;0
```

Associated `tcpdump dst port 80 -A` output (snipped):

```
User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3)
Connection: close
Host: www.example.com
Accept: */*
Cookie: foo=bar
```

Run with --hostname and a "Host" header with --header:

```
./check_http --hostname=www.example.com --header="Host: www.example.com"
HTTP WARNING: HTTP/1.1 400 Bad Request - 504 bytes in 0.039 second response time |time=0.039264s;;;0.000000 size=504B;;;0
```

*Note:  HTTP 400 response is expected since we sent two `Host:` headers.*

Associated `tcpdump dst port 80 -A` output (snipped):

```
User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3)
Connection: close
Host: www.example.com
Accept: */*
Host: www.example.com
```
What version are you running?  Can you replicate this and show your commands and their output?

Or are you saying that one of these settings should override the other?

**Comment by koenwtje** [comment]
Thanks for looking into this.

I'm running the 2.1.0-beta-RC1 branch, commit 8077cf0375.

    ./check_http --version
    check_http v2.0.3.121.g8077 (nagios-plugins 2.0.3)

I've got nc listening on 127.0.0.1:8080 (`nc -l 8080`). When I run

    ./check_http --hostname 'example.com' -p 8080 localhost

as expected, nc shows the following output:

    GET / HTTP/1.1
    User-Agent: check_http/v2.0.3.121.g8077 (nagios-plugins 2.0.3)
    Connection: close
    Host: example.com:8080
    Accept: */*

However, if I add a `--header` option, the 'Host:' header is dropped:

    ./check_http --hostname 'example.com' --header 'X-foo: bar' -p 8080 localhost

gives the following output:

    GET / HTTP/1.1
    User-Agent: check_http/v2.0.3.121.g8077 (nagios-plugins 2.0.3)
    Connection: close
    X-foo: bar
    Accept: */*

I tested again on the master branch, and I couldn't replicate the behavior there either. Sorry, I could've given this information (like version number) in the first place....

**Comment by koenwtje** [comment]
According to a `git bisect`, the issue was introduced in 388ea92804371c7e78b60fcc13623add1431dadc

**Comment by koenwtje** [comment]
Fixed by #87 of #82.

**Comment by abrist** [comment]
Fixed in commit: https://github.com/nagios-plugins/nagios-plugins/commit/8ab801b8d97cbedcec61c31071e795662015a72e
